database/scripts/lib/buildfarm_tools.rb

```diff
@@ -1,40 +1,116 @@
-def parse_sql_output(output, keys = [])
-  # This function parses the output of an SQL query and returns an array of hashes
-  # The keys of the hashes are the column names of the SQL query or the keys passed as parameter
-  # Note: The SQL query must be formatted using sql_run.sh command
+# frozen_string_literal: true
 
-  if output.empty?
-    return nil
+require 'open3'
+
+module BuildfarmToolsLib
+  class BuildfarmToolsError < RuntimeError; end
+
+  KNOWN_REASONS = Set.new([
+                            'Jenkins channel closing down (Agent disconnected)',
+                            'Network error: Failed to clone github repo'
+                          ])
+
+  CONSISTENT_THRESHOLD = 3
+  FLAKY_BUILDS_THRESHOLD = 3 # For 15 days
```

Comment:
Maybe you also want to define the 15 days as a constant? Something along the lines of `FLAKY_BUILDS_DAYS_RANGE`

Reply:
This is parametrized in all the code, so there is no hardcoded value for the flaky builds days range. I can add it as a default value instead